### PR TITLE
[#166584723] add test for PaymentProblemJson response decode

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🐞 [#166584723](https://www.pivotaltracker.com/story/show/166584723): Invalid PaymentFault nella risposta del pagopa-proxy-test

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/services/__tests__/pagoPAProxyService.test.ts [hunk 1]
```diff
@@ -177,6 +182,24 @@ describe("PagoPAProxyService#getPaymentInfo", () => {
     expect(res.kind).toEqual("IResponseErrorValidation");
   });
 
+  it("getPaymentInfoDefaultDecoder must validate correctly a PaymentProblemJson response", async () => {
```
Comment (cloudify) [inline src/services/__tests__/pagoPAProxyService.test.ts]: the test reads

`it getPaymentInfoDefaultDecoder must validate correctly a PaymentProblemJson response`

File: src/services/__tests__/pagoPAProxyService.test.ts [hunk 2]
```diff
@@ -177,6 +182,24 @@ describe("PagoPAProxyService#getPaymentInfo", () => {
...
+    const a = {
+      detail: PaymentFaultEnum.DOMAIN_UNKNOWN
+    };
+    const validator = getPaymentInfoDefaultDecoder();
+    const res = mockRes();
+    // tslint:disable: no-object-mutation
+    res.body = a;
+    res.status = 500;
+    // tslint:enable: no-object-mutation
+    const validation = await validator(res);
```
Comment (cloudify) [inline src/services/__tests__/pagoPAProxyService.test.ts]: does `validator` return a promise?

File: src/services/__tests__/pagoPAProxyService.test.ts [hunk 3]
```diff
@@ -177,6 +182,24 @@ describe("PagoPAProxyService#getPaymentInfo", () => {
     expect(res.kind).toEqual("IResponseErrorValidation");
   });
 
+  it("getPaymentInfoDefaultDecoder must validate correctly a PaymentProblemJson response", async () => {
+    const a = {
+      detail: PaymentFaultEnum.DOMAIN_UNKNOWN
+    };
+    const validator = getPaymentInfoDefaultDecoder();
+    const res = mockRes();
+    // tslint:disable: no-object-mutation
```
Comment (cloudify) [inline src/services/__tests__/pagoPAProxyService.test.ts]: don't disable globally, use `disable-next-line` instead

File: src/services/__tests__/pagoPAProxyService.test.ts [hunk 4]
```diff
@@ -177,6 +182,24 @@ describe("PagoPAProxyService#getPaymentInfo", () => {
...
+    res.body = a;
+    res.status = 500;
+    // tslint:enable: no-object-mutation
+    const validation = await validator(res);
+    if (validation !== undefined) {
```
Comment (cloudify) [inline src/services/__tests__/pagoPAProxyService.test.ts]: the decoder should never return `undefined` AFAIK

File: src/services/__tests__/pagoPAProxyService.test.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/services/__tests__/pagoPAProxyService.test.ts]: Yes it does. `export declare type ResponseDecoder<R> = (response: Response) => Promise<t.Validation<R> | undefined>;`

File: src/services/__tests__/pagoPAProxyService.test.ts (hunk 4 above)
Comment (BurnedMarshal) [reply src/services/__tests__/pagoPAProxyService.test.ts]: like [here](https://github.com/teamdigitale/io-backend/pull/477#discussion_r321669724)

File: src/services/__tests__/pagoPAProxyService.test.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/services/__tests__/pagoPAProxyService.test.ts]: fixed in 35258de. I don't now if now is better.

File: src/services/__tests__/pagoPAProxyService.test.ts (hunk 3 above)
Comment (BurnedMarshal) [reply src/services/__tests__/pagoPAProxyService.test.ts]: fixed in 35258de

**Pull request closed without merging**